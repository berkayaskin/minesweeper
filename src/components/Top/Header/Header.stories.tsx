import { Meta, Story } from '@storybook/react/types-6-0'
import { Header } from './Header'

import { HeaderProps } from './Header'

export default {
  title: 'Top/Header',
  component: Header,
} as Meta

const Template: Story<HeaderProps> = (args) => <Header {...args} />

export const GameHeader = Template.bind({})

GameHeader.args = {
  title: 'Minesweeper',
}
