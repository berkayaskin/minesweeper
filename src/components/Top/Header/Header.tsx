import { Heading } from '@chakra-ui/react'

export interface HeaderProps {
  /**
   * Title of the game
   * @example 'Minesweeper'
   * @example 'Sudoku'
   * @example 'Chess'
   */
  title: string
}

export const Header = ({ title }: HeaderProps) => {
  return (
    <Heading textAlign='center' fontFamily='monospace' mb='8' fontSize='5xl' color='whiteAlpha.800'>
      {title}
    </Heading>
  )
}
