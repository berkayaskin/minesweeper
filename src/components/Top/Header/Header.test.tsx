import { render } from '@testing-library/react'
import { Header } from './Header'

it('should renders the title correctly', () => {
  const { asFragment } = render(<Header title='Minesweeper' />)
  expect(asFragment()).toMatchSnapshot()
})
