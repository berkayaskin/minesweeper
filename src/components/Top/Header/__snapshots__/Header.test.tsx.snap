// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders the title correctly 1`] = `
<DocumentFragment>
  <h2
    class="chakra-heading css-6gnk2m"
  >
    Minesweeper
  </h2>
</DocumentFragment>
`;
